File: backend/app/schemas/appointment.py
from datetime import datetime
from typing import Optional

from pydantic import BaseModel

from app.models.enums import AppointmentStatus
from app.schemas.common import TimestampedModel


class AppointmentBase(BaseModel):
    """base appointment schema with business, client, and time window"""
    business_id: str
    client_id: str
    start_time: datetime
    end_time: datetime


class AppointmentCreate(AppointmentBase):
    """create appointment payload with optional notes"""
    notes: Optional[str] = None


class AppointmentUpdate(BaseModel):
    """update appointment payload with optional status and times"""
    status: Optional[AppointmentStatus] = None
    start_time: Optional[datetime] = None
    end_time: Optional[datetime] = None


class AppointmentRead(AppointmentBase, TimestampedModel):
    """complete appointment response with calendar integration info"""
    id: str
    status: AppointmentStatus
    notes: Optional[str] = None
    calendar_event_id: Optional[str] = None

File: frontend/lib/models/appointment.dart
class Appointment {
  const Appointment({
    required this.id,
    required this.businessId,
    required this.clientId,
    required this.startTime,
    required this.endTime,
    required this.status,
  });

  final String id;
  final String businessId;
  final String clientId;
  final DateTime startTime;
  final DateTime endTime;
  final String status;

  factory Appointment.fromJson(Map<String, dynamic> json) {
    return Appointment(
      id: json['id'] as String,
      businessId: json['business_id'] as String,
      clientId: json['client_id'] as String,
      startTime: DateTime.parse(json['start_time'] as String),
      endTime: DateTime.parse(json['end_time'] as String),
      status: json['status'] as String,
    );
  }
}
